packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -234,13 +234,27 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> {
     super.didChangeNext(nextRoute);
   }
 
+  bool _shouldJumpTrain(Animation<double> firstTrain, Animation<double> secondTrain) {
+    // We should jump to the other train if two trains will never cross in the
```

Comment:
The code below also doesn't seem to check that the trains will not cross in the future.

Reply:
The check is to detect whether they will cross with the current animations.

Reply:
The test below only looks at the current value and the direction of the animation. How's that enough to determine if they will cross? AFIK, the animation values doesn't have to run monotonically from 0.0 to 1.0. It could bounce up and down in between?